src/frontend/src/components/ui/TicketList.tsx

```diff
@@ -0,0 +1,247 @@
+/* eslint-disable no-console */
+import { ArrowRightIcon } from '@heroicons/react/24/solid';
+import { format } from 'date-fns';
+import Link from 'next/link';
+import React, { useEffect, useState } from 'react';
+import { ClipLoader } from 'react-spinners';
+import { toast, ToastContainer } from 'react-toastify';
+
+import { TicketListSkeleton } from '@/components/ui/Skeletons';
+
+import BrokenTicket from './NoTicketFoundView';
+
+const FETCH_SIZE = 150;
+const PAGE_SIZE = 10;
+
+type Ticket = {
+  ticket_id: string;
+  title: string;
+  reported_by: string;
+  assigned: string;
+  status: string;
+  priority: string;
+  created_at: string;
+};
+
+type TicketListProps = {
+  page: number;
+  setTotalPages: (pages: number) => void;
+  setDisableNext: (disable: boolean) => void;
+};
+
+const TicketList: React.FC<TicketListProps> = ({ page, setTotalPages, setDisableNext }) => {
+  const [data, setData] = useState<Ticket[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [isFinalBatch, setIsFinalBatch] = useState<boolean>(false);
+  const [loadingState, setLoadingState] = useState<{ [key: string]: boolean }>({}); // Track updates
+
+  const batchNumber = Math.floor((page - 1) / 15) + 1;
+  const customerGuid = '9a376cd0-396e-4a5a-9313-f8dfcfcba174'; // Replace with actual GUID
+
+  useEffect(() => {
+    const fetchTickets = async () => {
+      setLoading(true);
+      try {
+        const response = await fetch(
+          `http://localhost:8000/tickets/customer/${customerGuid}?page=${batchNumber}&page_size=${FETCH_SIZE}`,
```

Comment:
Lets move the API calls to src/frontend/src//api

Lets make it as a server side component 
and then import that component into this file

use env variables for host and port